Source/ACE.Entity/AceObject.cs

```diff
@@ -1817,8 +1841,7 @@ public object Clone(uint guid)
             AceObject ret = (AceObject)Clone();
             ret.AceObjectId = guid;
             // We are cloning a new AceObject with a new AceObjectID - need to set this to false. Og II
-            ret.HasEverBeenSavedToDatabase = false;
-            ret.IsDirty = true;
+            ret.SetDirtyFlags();
```

Comment:
it is not enough to set the flag on the top level.   This looked like it was working because until this work, we never really saved items to the database.   What I put in - I tested and it works - you have to do a deep set the setDirtyFlags that primes it to save to the database for the first time from a clone.   The flags are maintained automatically after that first save as items are changed.   It is a good system Mogwai put in, we just needed to debug the first real saves.   This does that.